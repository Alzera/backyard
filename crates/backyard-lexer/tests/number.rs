use backyard_lexer::lex_eval;

#[test]
fn basic() {
  let tokens = lex_eval("67").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn decimal() {
  let tokens = lex_eval("6.7").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn only_decimal() {
  let tokens = lex_eval(".67").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn hex() {
  let tokens = lex_eval("0xff9abc").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn binary() {
  let tokens = lex_eval("0b1010_1010").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}
