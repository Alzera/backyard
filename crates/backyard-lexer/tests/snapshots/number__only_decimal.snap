---
source: crates/backyard-lexer/tests/number.rs
assertion_line: 18
expression: tokens
snapshot_kind: text
---
- token_type: Number
  value: ".67"
  line: 1
  column: 0
  offset: 0
